eng/Generate.ps1

```diff
@@ -310,6 +299,26 @@ if (![string]::IsNullOrWhiteSpace($filter))
     }
 }
 
+if ($reset -or $env:TF_BUILD)
+{
+    $cadlCount = ([string]::IsNullOrWhiteSpace($filter) ? $cadlDefinitions : $cadlDefinitions.Keys.Where({$_ -match $filter})).Count
+    $swaggerCount = $keys.Count - $cadlCount
+    if ($swaggerCount -gt 0) 
+    {
+        AutoRest-Reset;
+    }
+    
+    if ($cadlCount -gt 0) 
```

Comment:
```suggestion
    move this into if (!$noBuild) block with the check $fast and $cadlCount
```

Reply:
!Fast path, we need to invoke generation of cadl.json and it need to re-build setup CADL

Reply:
> we need to invoke generation of cadl.json and it need to re-build setup CADL

Do we need to re-build CADL for every change in CADL input file?

Reply:
In CI we need to use the emitter and autorest.csharp built by the source code to generate SDK from Cadl. So we need to build the emitter and autorest.csharp once every time run Generate.ps1 if `-noBuild` is not specified.